aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,130 @@
+
+## Optimization
+
+Traditionally, .NET has focused on platforms where the app's binary size isn't a major concern. It doesn't really matter whether a server-side ASP.NET app is 1MB or 50MB. It's only a moderate concern for native desktop or mobile apps. But for browser apps, payload size is critical.
+
+Development efforts are aimed at reducing the download size of the Mono runtime and .NET app assemblies. Here are three phases of size optimization the Blazor engineering team has in mind:
+
+1. Mono runtime stripping
+
+   The Mono runtime contains many desktop-specific features. We hope that the Blazor packages will contain a trimmed version of Mono that is substantially smaller than the full-fat distribution. In an optimization experiment, the Mono runtime was pruned of unnecessary code. Over 70% of the Mono *wasm* file was removed while keeping a basic app working.
+
+1. Publish-time IL stripping
+
+   The .NET IL linker (originally based on the Mono linker) performs static analysis to determine which parts of .NET assemblies can ever get called by an app, then it strips out everything else.
+
+   This is equivalent to *tree shaking* in JavaScript, except the IL linker is much more fine-grained, operating at the level of individual methods. The IL Linker removes all of the system library code that the app isn't using, which often results in a 70% or greater reduction in code size.
+
+1. Compression
+
+   Most web servers support HTTP compression, which typically cuts the remaining payload size by a further 75%.
+
+Overall, a .NET-based browser app is never going to be as tiny as a minimal React app, but the goal is to make it small enough that a typical user with average Internet bandwidth won't notice or care about an app's first load time. After first load, the app's assemblies are fully cached.
+
+## Deployment
+
+Developers have the option of using Blazor for only client-side development or for full-stack .NET development. Full-stack development offers many advantages &mdash; client- and server-side development uses the same tooling, build infrastructure, and language. Code can be shared between client and server apps.
+
+For ASP.NET Core apps, [middleware](xref:fundamentals/middleware/index) will offer an easy path to serve a Blazor UI seamlessly from ASP.NET Core. Equally important are developers who don't yet use ASP.NET Core. To make Blazor a viable consideration for developers using Node.js, Rails, PHP, or even for serverless web apps, ASP.NET Core isn't required on the server. When a Blazor app is built, a *dist* directory is produced containing nothing but static files. The contents of the *dist* folder can be hosted on the Azure CDN, GitHub Pages, Node.js servers, and many other servers and services.
```

Comment:
~~For ASP.NET Core apps, [middleware](xref:fundamentals/middleware/index) will offer an easy path to serve a Blazor UI seamlessly from ASP.NET Core.~~
For ASP.NET Core, [middleware](xref:fundamentals/middleware/index) offers an easy path to serve a Blazor UI seamlessly from an ASP.NET Core app.